src/ui/utils/comments.ts

```diff
@@ -37,12 +37,11 @@ export function formatRelativeTime(date: Date) {
 }
 
 export function commentKeys(comments: (Comment | Reply)[]): string[] {
-  const createdAt = orderedByApproximatelyCreatedAt(comments);
-  return comments.map((c, i) => commentKey(c, createdAt[i]));
+  return comments.map((c, i) => commentKey(c));
 }
 
-function commentKey(comment: Comment | Reply, createdAtOrder: number): string {
-  return compact([createdAtOrder + 1, commentIdentifiers(comment)]).join("-");
+export function commentKey(comment: Comment | Reply): string {
+  return compact([comment.createdAt, commentIdentifiers(comment)]).join("-");
 }
```

Comment:
I am pretty sure that these keys are more stable when you use the createdAt field directly which I confirmed was not causing flickering

Reply:
Hmmmmmm... `createdAt` should change when persisting to the server, which is why I didn't use it originally.

Reply:
That's what I thought as well, but we actually don't update the Apollo representation when it comes back.